20241014-wordcloud/src/routes/+page.svelte

```diff
@@ -0,0 +1,170 @@
+
+    async function submitWord() {
+        const rgx = /drop\s+table/i;
+        if (rgx.test(submission)) {
+            alert("Nice try, but my SQL inputs are thoroughly sanitised.");
+        }
+
+        if (submission === "") {
+            return;
+        }
+
+        // Sanitise input
+        submission = submission
+            .toLowerCase()
+            .trim()
+            .split(/\s+/).join("-")
+            .replace(/[^a-z0-9-_.]/g, "");
+
+        if (submission === "") {
+            invalidate("invalid character in input");
+            return;
+        }
+
+        if (localStorage.getItem(`wordcloud___${submission}`) === "true") {
+            invalidate("already submitted");
+            return;
+        } else {
+            localStorage.setItem(`wordcloud___${submission}`, "true");
+        }
+
+        const response = await fetch("/answer", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({text: submission}),
+        });
+        words = await response.json();
+        timer = maxTimer;
+        submission = "";
+    }
+
+    async function getWords() {
+        const response = await fetch("/ask");
+        words = await response.json();
+        // Print to console so that it's easy to get hold of the data if we need to replot
+        let s = [];
+        for (const {text, count} of words) {
+            for (let i = 0; i < count; i++) {
+                s.push(text);
+            }
+        }
+        console.log(s.join(" "));
+    }
+
+    let timer = maxTimer;
+
+    onMount(async () => {
+        await getWords();
+        setInterval(async () => {
+            timer -= 1;
+            if (timer <= 0) {
+                await getWords();
+                timer = maxTimer;
+            }
+        }, 1000);
+    });
+
+    let totalWords: number;
+    $: {
+        totalWords = words.map((word) => word.count).reduce((a, b) => a + b, 0);
+    }
+</script>
+
+<svelte:head>
+    <title>A word cloud...</title>
+</svelte:head>
+
+<div id="wrapper-wrapper">
+    <h1>Hello, another REG survey</h1>
+
+    <div id="wrapper">
+        <div id="input-form">
+            <p>Please type in any programming languages or technologies you have used in your time in REG.</p>
+            <p>This can include not-entirely-serious-work things, such as Hack Week projects!</p>
+
+            <form on:submit|preventDefault={async () => submitWord()}>
+                <input id="submission" type="text" placeholder="Enter a word..."
+                    bind:this={submissionInput}
+                    bind:value={submission} />
+                <button id="submit">Submit</button>
+            </form>
+
+            <p>The word cloud on the right automatically updates whenever you submit a word, or {maxTimer} seconds after the last update, whichever comes first. <span class="smaller">(Next refresh in: {timer} seconds)</span></p>
```

Comment:
```suggestion
            <p>The word cloud automatically updates whenever you submit a word, or {maxTimer} seconds after the last update, whichever comes first. <span class="smaller">(Next refresh in: {timer} seconds)</span></p>
```
Given you can't control how the client will render things, for accessibility, I think it may be better not to refer to things by their position on the page. Having said that, I actually think this sentence may be unnecessary; better left out completely.